azerothcore/azerothcore-wotlk
# feat(Core/Disables): Implement DISABLE_TYPE_GAME_EVENT

## Changes Proposed:
- Allow disabling game events.
- Stop events if they're disabled while running during .reload disables.
- Disabled events can still be forced to start using .event start.

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:


```sql
DELETE FROM disables WHERE sourcetype = 9 AND entry = 26;
INSERT INTO `disables` (`sourceType`, `entry`, `flags`, `params_0`, `params_1`, `comment`) VALUES
(9, 26, 0, 0, 0, 'Disable Pilgrim\'s Bounty');
```

1. https://www.nirsoft.net/utils/run_as_date.html worldserver - 25th Nov or so
2. Check if the event starts
3.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
